// @reown/appkit-utils/dist/esm/src/solana/SolanaHelpersUtils.js
import { ConstantsUtil as CommonConstantsUtil } from '@reown/appkit-common';
import { SolConstantsUtil } from './SolanaConstantsUtil.js';
const NetworkImageIds = {
    '5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp': 'a1b58899-f671-4276-6a5e-56ca5bd59700',
    '4uhcVJyU9pJkvQyS88uRDiswHXSCkY3z': 'a1b58899-f671-4276-6a5e-56ca5bd59700',
    EtWTRABZaYq6iMfeYKouRu166VU2xqa1: 'a1b58899-f671-4276-6a5e-56ca5bd59700'
};
export const SolHelpersUtil = {
    detectRpcUrl(chain, projectId) {
        if (chain.rpcUrls.default.http[0]?.includes(new URL(CommonConstantsUtil.BLOCKCHAIN_API_RPC_URL).hostname)) {
            return `${chain.rpcUrls.default.http[0]}?chainId=solana:${chain.id}&projectId=${projectId}`;
        }
        return chain.rpcUrls.default.http[0];
    },
    getChain(chains, chainId) {
        const chain = chains.find(lChain => lChain.id === chainId);
        if (chain) {
            return chain;
        }
        return SolConstantsUtil.DEFAULT_CHAIN;
    },
    getChainFromCaip(chains, chainCaipId = ':') {
        const chainId = (chainCaipId?.split(':')[1] ?? '').replace(/\s/gu, '');
        const selectedChain = chains.find(chain => chain.id === chainId);
        if (selectedChain) {
            return {
                ...selectedChain,
                id: `solana:${chainId}`,
                imageId: NetworkImageIds[chainId],
                chainNamespace: CommonConstantsUtil.CHAIN.SOLANA
            };
        }
        return {
            ...SolConstantsUtil.DEFAULT_CHAIN,
            id: `solana:${chainId}`,
            imageId: NetworkImageIds[chainId],
            chainNamespace: CommonConstantsUtil.CHAIN.SOLANA
        };
    },
    hexStringToNumber(value) {
        const hexString = value.startsWith('0x') ? value.slice(2) : value;
        const decimalValue = parseInt(hexString, 16);
        return decimalValue;
    },
    getAddress(provider) {
        const address = provider.publicKey?.toBase58();
        return address;
    }
};
